packages/core/src/domain/tracekit/tracekit.ts

```diff
@@ -55,36 +57,44 @@ export function startUnhandledErrorCollection(callback: UnhandledErrorCallback)
  */
 function instrumentOnError(callback: UnhandledErrorCallback) {
   return instrumentMethodAndCallOriginal(window, 'onerror', {
-    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: Error) {
-      let stack: StackTrace
-
-      if (errorObj) {
-        stack = computeStackTrace(errorObj)
+    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: unknown) {
+      // Priority is given to errorObj if it is an instance of Error. This allows us to compute the most comprehensive stack trace.
+      if (errorObj instanceof Error) {
+        const stack = computeStackTrace(errorObj)
         callback(stack, errorObj)
       } else {
-        const location = {
-          url,
-          column: columnNo,
-          line: lineNo,
-        }
+        let name: string | undefined
+        let msg: string | undefined
 
-        let name
-        let msg = message
+        // Try to extract information about the error from the provided message
         if ({}.toString.call(message) === '[object String]') {
-          const groups = ERROR_TYPES_RE.exec(msg as string)
+          msg = message as string
+          const groups = ERROR_TYPES_RE.exec(message as string)
           if (groups) {
             name = groups[1]
             msg = groups[2]
           }
         }
 
-        stack = {
-          name,
-          message: typeof msg === 'string' ? msg : undefined,
-          stack: [location],
+        // If an errorObj is provided, give priority to its content to populate the message of the stack trace
+        if (errorObj) {
+          msg = jsonStringify(errorObj)
         }
 
-        callback(stack, message)
+        const stack: StackTrace = {
+          name,
+          // We only include the `Uncaught` prefix when errorObj is present, to preserve backward compatibility
+          message: errorObj && msg ? `${NonErrorPrefix.UNCAUGHT} ${msg}` : msg,
+          stack: [
+            {
+              url,
+              column: columnNo,
+              line: lineNo,
+            },
+          ],
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏when `url`, `columnNo` and `lineNo` is undefined, it won't provide anything meaningful. Maybe check wether some are defined before setting it? You might go even further and add the stack only if `url` is defined, because a line/column without url is not really helpful.